src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -840,6 +840,10 @@ public function findDefinition($id)
      */
     public function createService(Definition $definition, $id, $tryProxy = true)
     {
+        if ($definition instanceof DefinitionDecorator) {
```

Comment:
What about checking this in a more generic way (i.e. forbidding all `Definition` subclasses)?